lib/modules/manager/composer/artifacts.ts

```diff
@@ -29,17 +29,43 @@ import {
   getPhpConstraint,
   requireComposerDependencyInstallation,
 } from './utils';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import type { HostRuleSearch } from '../../../util/host-rules';
+
+function findGithubPersonnalAccessToken(
+  search: HostRuleSearch
+): string | undefined {
+  const token = hostRules.find(search)?.token?.replace('x-access-token:', '');
+  if (isPersonnalAccessToken(token)) {
+    return token;
+  }
+  return undefined;
+}
+
+function isPersonnalAccessToken(token: string): boolean {
+  return /^ghp_/.test(token);
+}
 
 function getAuthJson(): string | null {
   const authJson: AuthJson = {};
 
-  const githubCredentials = hostRules.find({
+  const githubToken = findGithubPersonnalAccessToken({
     hostType: PlatformId.Github,
     url: 'https://api.github.com/',
   });
-  if (githubCredentials?.token) {
+  if (githubToken) {
+    authJson['github-oauth'] = {
+      'github.com': githubToken,
+    };
+  }
+
+  const gitTagsGithubToken = findGithubPersonnalAccessToken({
+    hostType: GitTagsDatasource.id,
```

Comment:
```suggestion
    hostType: PackagistDatasource.id,
```

we should stay on `packagist` hostType.

Reply:
I took git-tags because that's what's used by the dependancy lookup phase  https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/composer/extract.ts#L136

I can change it for packagist (that was my first idea) but we will need to configure both datasource (packagist + git-tags) with the same PAT.